beacon-chain/monitor/service.go

```diff
@@ -40,9 +47,12 @@ type ValidatorMonitorConfig struct {
 // Service is the main structure that tracks validators and reports logs and
 // metrics of their performances throughout their lifetime.
 type Service struct {
-	config                *ValidatorMonitorConfig
-	latestPerformance     map[types.ValidatorIndex]ValidatorLatestPerformance
-	aggregatedPerformance map[types.ValidatorIndex]ValidatorAggregatedPerformance
+	config                      *ValidatorMonitorConfig
+	monitorLock                 sync.RWMutex
```

Comment:
it's not clear to me what this lock is protecting, `config`? `latestPerformance`? or everything? Perhaps a comment or better name would suffice